.github/workflows/main.yml

```diff
@@ -1,79 +1,256 @@
+
+      - name: install python
+        uses: actions/setup-python@v2
+
+      - name: install poetry2conda
+        run: pip install poetry2conda
 
-    - name: set up environment
-      run: ./ci/setup_env.sh "${{ matrix.python_version }}" "$BACKENDS"
-      shell: bash
+      - name: generate conda env file
+        run: |
+          set -euo pipefail
 
-    - name: run tests
-      run: PYTEST_BACKENDS=$BACKENDS ./ci/run_tests.sh --numprocesses auto
-      shell: bash
+          poetry2conda --dev --extras all pyproject.toml - | tee environment.yaml
 
-    - name: pytest errors
-      run: ./ci/pytest_errors.sh
-      if: failure()
+      - name: upload conda env file
+        uses: actions/upload-artifact@v2
+        with:
+          name: conda-env-file
+          path: environment.yaml
+          if-no-files-found: error
 
-  Tests_sql:
-    name: Tests SQL
+      - name: cleanup generated env files
+        run: rm environment.yaml
+
+  test_no_backends:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - gen_conda_env_file
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - "3.7"
+          - "3.9"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: download env file
+        uses: actions/download-artifact@v2
+        with:
+          name: conda-env-file
+          path: tmp
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          mamba-version: "*"
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          channel-priority: strict
+          activate-environment: ibis
+          python-version: ${{ matrix.python-version }}
+          environment-file: tmp/environment.yaml
+          condarc-file: ci/condarc
+
+      - name: install ibis
+        run: pip install .
+
+      - name: run tests
+        run: |
+          pytest \
+            ibis/tests \
+            --numprocesses auto \
+            --quiet \
+            -ra \
+            --junitxml=junit.xml \
+            --cov=ibis \
+            --cov-report=xml:coverage.xml | tee pytest.log
+
+      - name: pytest errors
+        run: ./ci/pytest_errors.sh
+        if: failure()
+
+  test_simple_backends:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - gen_conda_env_file
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - "3.7"
+          - "3.9"
+        pyarrow:
+          - "1"
+          - "5"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: download env file
+        uses: actions/download-artifact@v2
+        with:
+          name: conda-env-file
+          path: tmp
+
+      - uses: conda-incubator/setup-miniconda@v2
```

Comment:
Same here, we don't have any dependencies requiring conda to be installed.

Reply:
Yep

Reply:
Done.